crates/ott-common/src/discovery/dns.rs

```diff
@@ -50,3 +63,26 @@ impl ServiceDiscoverer for DnsServiceDiscoverer {
         DiscoveryMode::Polling(Duration::from_secs(10))
     }
 }
+
+mod test {
+    #[allow(unused_imports)]
+    use crate::discovery::DnsDiscoveryConfig;
+    #[allow(unused_imports)]
+    use serde_json::json;
+    #[allow(unused_imports)]
+    use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+
+    #[tokio::test]
+    async fn server_deserializes_correctly() {
+        let json = json!({
+            "service_port": 8080,
+            "dns_server": "127.0.0.1:8080",
```

Comment:
Does this mean that the port is required? Typically dns is served on port 53 and we should default to that if the user doesn't provide it.

Reply:
it seems like the port is required for creating a socketaddr struct: https://doc.rust-lang.org/std/net/enum.SocketAddr.html

Reply:
I would much prefer we don't require the port to be specified. Like I said before, we should default to 53 if the user doesn't provide it.

This will likely require customizing how serde deserializes this field. Take a look at:
- https://brokenco.de/2020/08/03/serde-deserialize-with-string.html
- https://serde.rs/custom-date-format.html

Keep in mind you only need to implement the deserialize side.

Reply:
I think this should be filed as a separate task:

1. The dns_server is customizable and this wasn't a requirement mentioned in the original issue
2. (I don't think) there was any way I could have reasonably anticipated the port needing to be specified would be a problem
3. For the sake of sprint planning
4. (imo) This just kinda feels like a separate thing

Completely understand if you don't want to merge this PR yet though. I'd be happy to hop on a call tomorrow or talk about it more on Thursday if you wanted.

Reply:
@coderabbitai can you open an issue to make the port for `dns_server` optional?